packages/tools/src/tools/segmentation/BrushTool.ts

```diff
@@ -69,7 +69,7 @@ class BrushTool extends BaseTool {
   private _editData: {
     segmentsLocked: number[]; //
     segmentationRepresentationUID?: string;
-    imageIdReferenceMap?: Map<string, string>;
+    imageIdReferenceMap?: Map<string, Set<string>>;
```

Comment:
It really isn't obvious to me how you would figure out which one is of interest.  I think they should be keyed by the segmentation UID, which would mean you get exactly one value, so that you would then undo this line.